### DISCUSSION TITLE
How to baseline without reset

### DISCUSSION THREAD

**Discussion by jsw** [prisma/prisma#15700] [Q&A] [answered]
I have a `schema.prisma` that has been applied to mysql dev and prod databases via `prisma db push`.
I don't have any migrations files or tables.
I could reset the dev database but not the prod database.
I tried following the instructions to generate a baseline but get the following result.

```
prisma migrate dev --name initial-migration --create-only

Drift detected: Your database schema is not in sync with your migration history.
...
✔ We need to reset the MySQL database "x" at "xxx:3306".
Do you want to continue? All data will be lost. … no
```

I tried running `prisma db pull` first, but got the same result when running `prisma migrate dev`.
How should I proceed?

**Comment by austincrim** [comment, marked as answer]
Hey @jsw 👋,

Do you have a `_prisma_migrations` table in your database? Could you query it and provide me with the result?

**Reply by jsw** [reply]
Hi @austincrim,

I do not have that table in either database. I messed up and have never run migrations. I prototyped with db push and also used that to create the production database, which is now populated with production data. Now that I need to make a schema change, I'm stuck. If it makes things easier I can reset the dev database, though it would be nice to not need to as it will give me confidence that I can safely baseline production. I can also create another temporary fresh development database to work off if that makes things easier to bootstrap this process.

**Reply by austincrim** [reply]
Do you have a local `migrations` folder?

**Reply by jsw** [reply]
No `migrations` folder

**Reply by austincrim** [reply]
Based off [this docs guide](https://www.prisma.io/docs/guides/database/developing-with-prisma-migrate/baselining#baselining-a-database), you can try creating an initial migration on your **development** database (this will require a reset) and then following the instructions in the doc to baseline your production database.

After you generate an initial migration using your dev database, you can run `prisma migrate resolve --applied name_of_initial_migration` against your **production** database, which should baseline it for future use.

**Reply by jsw** [reply]
Do I need to or should I do a `prisma db pull`? I notice that it reformats my file and adds several `@@index` rows. Slightly unnerving. Not sure the consequence of updating my `schema.prisma` like this.

**Reply by jsw** [reply]
Also confusing to see these messages. My models already have all these `@@map`, `@default(uuid())` and `@updatedAt`

```
// *** WARNING ***
// These models were enriched with `@@map` information taken from the previous Prisma schema.
...
// These id fields were enriched with `@default(uuid())` information taken from the previous Prisma schema.
...
// These DateTime fields were enriched with `@updatedAt` information taken from the previous Prisma schema.

**Reply by pantharshit00** [reply]
Hello @jsw,

You can try this workaround here: https://github.com/prisma/prisma/issues/11555#issuecomment-1027242342

It won't prompt for a reset and simply tell migrate that initial migration was already applied

**Reply by jsw** [reply]
Thanks @austincrim and @pantharshit00 for your advice. I ended up doing the following. Since I actually have 2 non-production databases to work with (personal dev and shared dev), I decided to reset the personal dev one and then treat the shared dev as production so that I had confidence that whatever I do to that one will work the same in production. 

I'm still a little uncertain about the addition of `@@index()` elements in my schema.prisma after the `db pull`, but hopefully it will all work out. Once again, I acknowledge I messed up in going to production without setting up migrations first.

on personal dev database
```
prisma db pull
prisma migrate dev --name initial-migration --create-only
prisma migrate dev
```

on shared dev database
```
prisma migrate resolve --applied 20221012161028_initial_migration
```

on prod database
```
prisma migrate resolve --applied 20221012161028_initial_migration
```

**Reply by zFlxw** [reply]
Thank you for clarifying what you did to resolve it. I had the exact same problem and due to your description, I was able to fix it. :)
